runtime/Dart/lib/src/atn/src/atn_config_set.dart

```diff
@@ -100,28 +103,34 @@ class ATNConfigSet extends Iterable<ATNConfig> {
   ///
   /// <p>This method updates {@link #dipsIntoOuterContext} and
   /// {@link #hasSemanticContext} when necessary.</p>
-  bool add(ATNConfig config,
-      [Map<Pair<PredictionContext, PredictionContext>, PredictionContext>
-          mergeCache]) {
+  bool add(
+    ATNConfig config, [
+    Map<Pair<PredictionContext, PredictionContext>, PredictionContext>?
+        mergeCache,
+  ]) {
     if (readOnly) throw StateError('This set is readonly');
     if (config.semanticContext != SemanticContext.NONE) {
       hasSemanticContext = true;
     }
     if (config.outerContextDepth > 0) {
       dipsIntoOuterContext = true;
     }
-    final existing = configLookup.lookup(config) ?? config;
+    final existing = configLookup!.lookup(config) ?? config;
```

Comment:
we're "sure" configLookup exists, because it is only null when is on readonly mode.